### ISSUE TITLE
default skin in manjaro-i3

### ISSUE THREAD

**Issue by oberon-manjaro** [Boruch-Baum/morc_menu#7]
Hi Boruch!
I just tried building a new i3 test ISO with morc_menu, wmutils and xdotool installed.
When I launch morc_menu it doesn't look what I expected, but like this:

![morc](https://cloud.githubusercontent.com/assets/11800714/14105975/fe7666b6-f5ae-11e5-8bff-cc54dfcfb45c.png)

while the config is containing (uncommented of course)
`external_skin_definition_file=${HOME}/.dmenurc`
aswell as
`use_mouse_position=TRUE`
Changing prefix/suffix in the config file does work. So the file is being read obviously...

The good news is, it launches without any delay :wink:

**Comment by Boruch-Baum** [comment]
Hi. I owe you an off-line message regarding a few other things in progress. For this issue, the most important line in the config file is "menu_cmd=". If you're using any relatively current github download of morc_menu, the default for that config variable WILL display as your screenshot illustrates. The config file includes four examples for that variable, the second of which is identified as exploiting all the extra features of the fully patched dmenu.

This is not an issue for the version of morc_menu being packaged for manjaro i3. For that linux distribution, and potentially for others, while the software is identical and the initial config file is identical, the package manager makes alterations to the default to match the perferences of the distribution developer, (eg, you). For manjaro, the PKGBUILD includes the commands to alter the default config file so the user need not alter anything in order to get the desired look.

NOW, as for your good news: I want to track down that issue, but since that's been opened as a separate issue / thread, I'll write something there for your comment.

Let me know if this issue is just a wrong menu_cmd setting. If not, please post your config file, using: sed -n '/^[a-z]/p' morc_menu_v1.conf > post_this_file

**Comment by oberon-manjaro** [comment]
Like I said, this happens on a freshly built manjaro-i3 ISO. So it uses the package now in manjaro repos.

```
external_skin_definition_file=${HOME}/.dmenurc
use_mouse_position=TRUE
x_position=300
y_position=20
line_height=20
avg_char_width=9
avg_err_char_width=10
menu_width=350
err_menu_width=350
menu_cmd="dmenu -i -l MENU_LINES "
error_cmd="dmenu -i -l 40 -nb red -nf black -fn DejaVu"
menu_prefix=" [menu]: "
menu_suffix=" >"
align_suffix=TRUE
desired_categories="Favorites Settings Development Documentation Education System Network Utility Graphics Office AudioVideo"
declare -A category_aliases=( [Utility]=Accessories [AudioVideo]=Multimedia [Network]=Internet )
unwanted_names=( )
unwanted_execs=( )
unwanted_specifics=( )
exclude_from_static="FALSE"
exclude_from_dynamic="TRUE"
additional_entries=( )
add_to_static="FALSE"
add_to_dynamic="TRUE"
max_num_backups=5
exclude_about="FALSE"
ignore_system_txt="FALSE
```

**Comment by oberon-manjaro** [comment]
Here's what the latest default **.dmenurc** looks like:

```
#
# ~/.dmenurc
#

## define the font for dmenu to be used
DMENU_FN="Cantarell-10"

## background colour for unselected menu-items
DMENU_NB="#002B36"

## textcolour for unselected menu-items
DMENU_NF="#EEE8D5"

## background colour for selected menu-items
DMENU_SB="#EEE8D5"

## textcolour for selected menu-items
DMENU_SF="#002B36"

## command for the terminal application to be used:
TERMINAL_CMD="terminal -e"

## export our variables
DMENU_OPTIONS="-fn $DMENU_FN -nb $DMENU_NB -nf $DMENU_NF -sf $DMENU_SF -sb $DMENU_SB"
```

**Comment by Boruch-Baum** [comment]
I looked at your config file, and I had it nailed: it's the line: "menu_cmd=". Look in the config file for the commented-out examples of alternatives.

```
# 3] Using a variable defined in an external_skin_definition_file,
#    in this case ${HOME}/.dmenurc
#      menu_cmd="dmenu -i -l MENU_LINES -x X_POSITION -y Y_POSITION -w MENU_WIDTH ${DMENU_OPTIONS}"
```

A separate issue is why the manjaro i3 iso PKGBUILD hasn't automatically changed the default config, but that's a discussion for the manjaro forum, not here.

**Comment by oberon-manjaro** [comment]
After our discussion I was expecting the config to be adjusted to manjaro-i3 automatically. But no problem. So I have commanted out `menu_cmd=` and adjusted prefix and suffix manually.
However, no matter if `use_mouse_position=TRUE` is commented or not or set to FALSE, the menu is always being spawned to mouse position...

**Comment by Boruch-Baum** [comment]
You're right, and it should, when the Manjaro PKGBUILD works as written! As mentioned on the Manjaro forum, someone incorrectly added a redirect, which broke that code. I'm leaving this issue open in order to look at your report about `use_mouse_position=TRUE`.

**Comment by Boruch-Baum** [comment]
I think I understand what might be happening. As per the documentation, `use_mouse_position` requires the optional dependencies of packages 'xdotool' and 'wmutils'. When your PKGBUILD was aborted because of the bad redirect, I suspect it also failed to install those. Try the following from the command line:

```
type -f xdotool wattr lsw
```

or just

```
pacman -Q xdotool wmutils
```

If those packages are missing, let me know how it goes after you install them. The PKGBUILD writer for morc_menu reports that he uploaded a new version to correct the redirect typo, so maybe better to de-install morc_menu and then re-install with the new package.

**Comment by oberon-manjaro** [comment]
Nono. The two packages are installed. morc_menu was already removed and reinstalled with the updated install script. The package is not in the repo, yet, but I am testing it locally...

**Comment by Boruch-Baum** [comment]
I can't reproduce what you're experiencing when those packages are installed. Have you run the two commands (`type -f` & `pacman -Q`) to verify the actual presence of those packages. Maybe also check the permissions?

**Comment by oberon-manjaro** [comment]
Ok. I found something:
The setting for `use_mouse_position=` at user level is only being respected when `use_mouse_position=TRUE` is commented in **/usr/share/morc_menu/** while the settings for pre-/suffix at user level are always being respected ;)

**Comment by oberon-manjaro** [comment]
@Boruch-Baum, any progress on this?
